docs/developer-docs.md

```diff
@@ -0,0 +1,96 @@
+EasyEngine Developer Docs
+=========================
+
+## Requirements
+
+* Git
+* PHP
+* Composer
+
+Before starting refer the [core repository structure](core-repo-structure.md) and the [structure for multiple repositories](structure-for-multiple-repos.md).
+
+## Steps for working on core repository
+
+1. Fork the EasyEngine core repository, clone it locally and checkout the development branch.
+```bash
+git clone git@github.com:your-username/easyengine.git && git checkout develop-v4 
+```
+2. Run `composer install` in the core repository after this.  
+
+3. Make required changes and check them by running it locally using the following from the easyengine repository root:
+
+```bash
+$ ./bin/ee command
+```
+
+## Steps for working on existing commands
+
+1. Clone the EasyEngine core repository locally and checkout the development branch.
+```bash
+git clone git@github.com:EasyEngine/easyengine.git && git checkout develop-v4 
+```
+
+2. Fork the command you want to work on.
+
+3. Update the `composer.json` in the EasyEngine core repository, replace that command from `require` block to have your command name. For example lets replace site command: 
+
+~~```"easyengine/site-command": "dev-master"```~~
+
+```"your-username/site-command": "dev-master"```
+
+4. Append the following section in the `composer.json` for development:
+```
+"repositories": {
+    "author/command-name": {
+        "type": "path",
+        "url": "path/to/your/forked/repository"
+    }
+}
+```
+
+5. Run `composer update` in the core repository after this.
+
+6. Then, run your commands locally using the following from the easyengine repository root:
+```bash
+$ ./bin/ee command
+```
+
+## Steps for creating a new command
+
+1. Fork the [command template repository](https://github.com/EasyEngine/command-template) and rename it to the command you want to create. This will now look like `author/command-name` in your github.
```

Comment:
Instead of fork, one can clone it and get started locally.
They'll not have to add any git remote or anything. Just specify a local path in composer.json

I believe for development this is better.